analyzer/emerald/incoming.go

```diff
@@ -349,38 +413,76 @@ func extractRound(b *block.Block, txrs []*sdkClient.TransactionWithResults, logg
 							if err2 != nil {
 								return fmt.Errorf("from: %w", err2)
 							}
+							eventRelatedAddresses = append(eventRelatedAddresses, fromAddr)
 							registerTokenDecrease(blockData.TokenBalanceChanges, eventAddr, fromAddr, amount)
 						}
 						if !bytes.Equal(toEthAddr, common.ZeroEthAddr) {
 							toAddr, err2 := registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, toEthAddr)
 							if err2 != nil {
 								return fmt.Errorf("to: %w", err2)
 							}
+							eventRelatedAddresses = append(eventRelatedAddresses, toAddr)
 							registerTokenIncrease(blockData.TokenBalanceChanges, eventAddr, toAddr, amount)
 						}
 						blockData.PossibleTokens[eventAddr] = &modules.EVMPossibleToken{
 							PossibleTypes: []modules.EVMTokenType{modules.EVMTokenTypeERC20},
 							EthAddr:       event.Address,
 						}
+						evmLogName := apiTypes.Erc20Transfer
+						evmLogParams := map[string]string{
```

Comment:
I've realized ... the order of args matters. We could introduce a separate structure that lists the args (and we could do it later too; for erc20 Transfer, everybody knows what's up), but I think it's a nicer interface to emit a single structure: instead of `map[string]string`, we can have `[]EvmEventArg`, with `type EvmEventArg struct { name: string, value: any, evm_type: string }`. That's my best guess on what we'll be able to extract and what will be relevant.

Differences:
- order is preserved because of the list
- value type is `any`. This should be the case anyway, even in the map. EVM supports [various types (int256, string, address, ...)](https://docs.soliditylang.org/en/v0.8.17/types.html#value-types) for event values. 
- a new attribute, `type`. We could even omit it for now, until we start supporting more kinds of events. It's a good idea to include it eventually probably because there's not a 1:1 mapping between JSON types (= what we're sending out) and EVM types.

I guess down the road we'll want to expose all event metadata that the ABI gives us; see [our ERC-20 example](https://github.com/oasisprotocol/oasis-indexer/blob/3c756580ea3fe1d7d5949b4062c30693074d8ad8/analyzer/evmabi/contracts/artifacts/ERC20.json#L230-L251) and [its documentation](https://docs.soliditylang.org/en/v0.8.17/abi-spec.html#events). But it's too early to worry about every fine detail, as long as we have an initial structure that's flexible enough that we can keep adding and exposing more properties. `EvmEventArg` is my attempt at striking that balance.

TLDR: Let's go with `[]EvmEventArg` here, and revisit in a few months if needed.

Reply:
Got it; updated